cmd/bosun/sched/check.go

```diff
@@ -508,22 +512,42 @@ func (s *Schedule) findUnknownAlerts(now time.Time, alert string) []models.Alert
 	return keys
 }
 
-func (s *Schedule) CheckAlert(T miniprofiler.Timer, r *RunHistory, a *conf.Alert) {
+func (s *Schedule) CheckAlert(T miniprofiler.Timer, r *RunHistory, a *conf.Alert) (cancelled bool) {
 	slog.Infof("check alert %v start", a.Name)
 	start := utcNow()
 	for _, ak := range s.findUnknownAlerts(r.Start, a.Name) {
 		r.Events[ak] = &models.Event{Status: models.StUnknown}
 	}
 	var warns, crits models.AlertKeys
-	d, err := s.executeExpr(T, r, a, a.Depends)
+	type res struct {
+		results *expr.Results
+		error   error
+	}
+	rc := make(chan res)
```

Comment:
Trying to reason about orphaned goroutines here. I think this chan should be buffered with size 1.